source/steamshim/src/child/child.cpp

```diff
@@ -0,0 +1,202 @@
+#include <cstdint>
+#include <cstring>
+#include <stdio.h>
+#define DEBUGPIPE 1
+#include "child_private.h"
+#include "../steamshim.h"
+#include "../steamshim_private.h"
+#include "../os.h"
+#include "../steamshim_types.h"
+
+int GArgc = 0;
+char **GArgv = NULL;
+
+static int initPipes(void)
+{
+    char buf[64];
+    unsigned long long val;
+
+    if (!getEnvVar("STEAMSHIM_READHANDLE", buf, sizeof (buf)))
+        return 0;
+    else if (sscanf(buf, "%llu", &val) != 1)
+        return 0;
+    else
+        GPipeRead = (PipeType) val;
+
+    if (!getEnvVar("STEAMSHIM_WRITEHANDLE", buf, sizeof (buf)))
+        return 0;
+    else if (sscanf(buf, "%llu", &val) != 1)
+        return 0;
+    else
+        GPipeWrite = (PipeType) val;
+    
+    return ((GPipeRead != NULLPIPE) && (GPipeWrite != NULLPIPE));
+} /* initPipes */
+
+static STEAMSHIM_Event* ProcessEvent(){
+    static STEAMSHIM_Event event;
+    // make sure this is static, since it needs to persist between pumps
+    static pipebuff_t buf;
+
+    if (!buf.Recieve())
+        return NULL;
+
+    if (!buf.hasmsg)
+        return NULL;
+
+    volatile unsigned int msglen =buf.ReadInt();
+
+
+    char type = buf.ReadByte();
+    event.type = (STEAMSHIM_EventType)type;
+
+    #if DEBUGPIPE
+    if (0) {}
+    #define PRINTGOTEVENT(x) else if (type == x) printf("Child got " #x ".\n")
```

Comment:
If you make this a switch without default case, you'll get a warning if you forget to implement a case.